src/ReportsSync.php

```diff
@@ -9,197 +9,40 @@
-		add_action( self::ORDERS_DELETE_BATCH_ACTION, array( __CLASS__, 'orders_lookup_delete_batch' ), 10, 4 );
-		add_action( self::ORDERS_DELETE_BATCH_INIT, array( __CLASS__, 'orders_lookup_delete_batch_init' ), 10, 3 );
-		add_action( self::SINGLE_ORDER_IMPORT_ACTION, array( __CLASS__, 'orders_lookup_import_order' ) );
-	}
-
-	/**
-	 * Regenerate data for reports.
-	 *
-	 * @param int|bool $days Number of days to import.
-	 * @param bool     $skip_existing Skip exisiting records.
-	 * @return string
-	 */
-	public static function regenerate_report_data( $days, $skip_existing ) {
-		if ( self::is_importing() ) {
-			return new \WP_Error( 'wc_admin_import_in_progress', __( 'An import is already in progress.  Please allow the previous import to complete before beginning a new one.', 'woocommerce-admin' ) );
-		}
-
-		self::reset_import_stats( $days, $skip_existing );
-		self::customer_lookup_import_batch_init( $days, $skip_existing );
-		self::queue_dependent_action( self::ORDERS_IMPORT_BATCH_INIT, array( $days, $skip_existing ), self::CUSTOMERS_IMPORT_BATCH_ACTION );
-
-		return __( 'Report table data is being rebuilt.  Please allow some time for data to fully populate.', 'woocommerce-admin' );
-	}
-
-	/**
-	 * Update the import stat totals and counts.
-	 *
-	 * @param int|bool $days Number of days to import.
-	 * @param bool     $skip_existing Skip exisiting records.
-	 */
-	public static function reset_import_stats( $days, $skip_existing ) {
-		$totals = self::get_import_totals( $days, $skip_existing );
-		update_option( 'wc_admin_import_customers_count', 0 );
-		update_option( 'wc_admin_import_orders_count', 0 );
-		update_option( 'wc_admin_import_customers_total', $totals['customers'] );
-		update_option( 'wc_admin_import_orders_total', $totals['orders'] );
-
-		// Update imported from date if older than previous.
-		$previous_import_date = get_option( 'wc_admin_imported_from_date' );
-		$current_import_date  = $days ? date( 'Y-m-d 00:00:00', time() - ( DAY_IN_SECONDS * $days ) ) : -1;
-
-		if ( ! $previous_import_date || -1 === $current_import_date || new \DateTime( $previous_import_date ) > new \DateTime( $current_import_date ) ) {
-			update_option( 'wc_admin_imported_from_date', $current_import_date );
-		}
 	}
 
 	/**
-	 * Get the import totals for customers and orders.
+	 * Get classes for syncing data.
 	 *
-	 * @param int|bool $days Number of days to import.
-	 * @param bool     $skip_existing Skip exisiting records.
 	 * @return array
 	 */
-	public static function get_import_totals( $days, $skip_existing ) {
-		$orders         = self::get_orders( 1, 1, $days, $skip_existing );
-		$customer_roles = apply_filters( 'woocommerce_analytics_import_customer_roles', array( 'customer' ) );
-		$customer_query = self::get_user_ids_for_batch(
-			$days,
-			$skip_existing,
+	public static function get_schedulers() {
+		return apply_filters(
```

Comment:
Rather than immediately returning here, this could be a good place to do interface checking as mentioned in other comments.

Reply:
I've added a check to make sure the parent class is `ImportScheduler` for all schedulers added here.  That will ensure we have all of the methods called that you mentioned in other comments without needing to check for their existence every time.